pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -314,6 +375,25 @@ func newIdentity(userAcc *toolchainv1alpha1.UserAccount, user *userv1.User) *use
 	return identity
 }
 
+func newNSTemplateSet(userAcc *toolchainv1alpha1.UserAccount) *toolchainv1alpha1.NSTemplateSet {
+	name := ToNSTemplateSetName(userAcc.Name)
+	nsTeplSet := &toolchainv1alpha1.NSTemplateSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: userAcc.Namespace,
+		},
+		Spec: toolchainv1alpha1.NSTemplateSetSpec{
+			TierName:   userAcc.Spec.TierName,
+			Namespaces: []toolchainv1alpha1.Namespace{},
+		},
+	}
+	return nsTeplSet
+}
+
 func ToIdentityName(userID string) string {
 	return fmt.Sprintf("%s:%s", config.GetIdP(), userID)
 }
+
+func ToNSTemplateSetName(name string) string {
+	return name
```

Comment:
We will probably need to do some validation of the username here. Not sure if we always can use the username as the namespace name + some prefix (stage, ide, etc). Can we exceed the namespace name length? Can username contain symbols that are not allowed in namespace names (probably not because both of them are k8s resource names), etc.